outputs/nexusiq.go

```diff
@@ -0,0 +1,181 @@
+	return nil
+}
+func (nexus *NexusIqOutput) auth() string {
+	return base64.StdEncoding.EncodeToString([]byte(nexus.User + ":" + nexus.Password))
+}
+
+func (nexus *NexusIqOutput) execute(method string, url string, payload string, headers map[string]string) (map[string]interface{}, error) {
+	client := http.DefaultClient
+
+	var reader io.Reader = nil
+
+	if payload != "" {
+		reader = strings.NewReader(payload)
+	}
+
+	req, err := http.NewRequest(method, url, reader)
+	if err != nil {
+		return nil, err
+	}
+	for name, value := range headers {
+		req.Header.Add(name, value)
+
+	}
+	req.Header.Add("Authorization", "Basic "+nexus.auth())
+
+	resp, err := client.Do(req)
+	if err != nil {
+		return nil, err
+	}
+
+	defer resp.Body.Close()
+
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return nil, err
+	}
+
+	if resp.StatusCode < 200 || resp.StatusCode > 299 {
+		msg := "received incorrect response status: %d. Body: %s"
+		return nil, fmt.Errorf(msg, resp.StatusCode, body)
+	}
+
+	r := make(map[string]interface{})
+
+	err = json.Unmarshal([]byte(body), &r)
+
+	if err != nil {
+		return nil, err
+	}
+
+	return r, nil
+
+}
+
+func (nexus *NexusIqOutput) getAppByNameAndOrg(organizationId string, appName string) (string, error) {
+	sanitizedAppName := sanitizedAppName(appName)
+	url := fmt.Sprintf("%s/api/v2/applications/organization/%s", nexus.Url, organizationId)
+	r, err := nexus.execute("GET", url, "", map[string]string{"Content-Type": "application/json"})
+	if err != nil {
+		return "", fmt.Errorf("error fetching application: %w", err)
+	}
+	applications := r["applications"].([]interface{})
+	for _, item := range applications {
+		app := item.(map[string]interface{})
+		if app["publicId"].(string) == sanitizedAppName {
+			return app["id"].(string), nil
+		}
+	}
+	return "", nil
+}
+func (nexus *NexusIqOutput) createApp(organizationId string, appName string) (string, error) {
+	sanitizedAppName := sanitizedAppName(appName)
+	payload := map[string]string{
+		"publicId":       sanitizedAppName,
+		"name":           sanitizedAppName,
+		"organizationId": organizationId,
+	}
+
+	b, err := json.Marshal(payload)
+	if err != nil {
+		return "", err
+	}
+
+	url := fmt.Sprintf("%s/api/v2/applications", nexus.Url)
+
+	r, err := nexus.execute("POST", url, string(b), map[string]string{"Content-Type": "application/json"})
+
+	if err != nil {
+		return "", fmt.Errorf("error creating application: %w", err)
+	}
+
+	return r["id"].(string), nil
+}
+
+func (nexus *NexusIqOutput) createOrGetApp(appName string) (string, error) {
+	app, err := nexus.getAppByNameAndOrg(nexus.OrganizationId, appName)
+	if err != nil {
+		return "", err
+	}
+	if app == "" {
+		app, err = nexus.createApp(nexus.OrganizationId, appName)
+		if err != nil {
+			return "", err
+		}
+	}
+	return app, nil
+}
+func (nexus *NexusIqOutput) registerBom(appId string, bom string) error {
+	url := fmt.Sprintf("%s/api/v2/scan/applications/%s/sources/cyclone", nexus.Url, appId)
```

Comment:
For my understanding: what endpoint is this using? Could you share the API docs?

Reply:
https://help.sonatype.com/iqserver/automating/rest-apis/third-party-scan-rest-api---v2

Reply:
thanks - can we add a doc somewhere that can document what versions of sbom are supported? XML/JSON?

Reply:
we can, but it's kind of implementation details. If customer use solution as is - this info doesn't help him, if he starts modifying template then there is direct reference to version.
https://github.com/aquasecurity/postee/blob/issue-99/rego-templates/vuls-cyclonedx.rego#L8-L9

Reply:
I don't mind to add it somewhere but I'm not sure about context of this information

Reply:
Yeah I see your point. I think we should add this as part of a bigger story, I added a task for it https://github.com/aquasecurity/postee/issues/335